collector/cluster_license_test.go

```diff
@@ -0,0 +1,102 @@
+// Copyright 2023 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package collector
+
+import (
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/go-kit/log"
+	"github.com/prometheus/client_golang/prometheus/testutil"
+)
+
+func TestIndicesHealth(t *testing.T) {
+	// Testcases created using:
+	//  docker run -d -p 9200:9200 elasticsearch:VERSION
+	//  curl http://localhost:9200/_license
+	tests := []struct {
+		name string
+		file string
+		want string
+	}{
+		{
+			name: "basic",
+			file: "../fixtures/clusterlicense/basic.json",
```

Comment:
Can we add versions to these tests? A different version of elasticsearch could break the format so it would be helpful to have versioned test data and tests.